src/Microsoft.Health.Extensions.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -51,6 +57,25 @@ public static void RegisterAssemblyModules(this IServiceCollection collection, A
         }
     }
 
+    public static ICollection<string> CheckIssues(this IServiceCollection collection)
+    {
+        var issues = new List<string>();
+        var multipleRegisteredServices = collection
+            .GroupBy(x => (x.ServiceType, x.ImplementationType, x.ImplementationInstance, x.ImplementationFactory))
```

Comment:
Thanks for the feedback here. Based on the comments, especially the 3rd bullet, I should probably remove the code altogether. What started all of this was a local debug session of fhir-server hitting Debug.Asserts in TypeRegistrationBuilder.AsImplementedInterfaces() . The debug message pointed to missing AsSelf() in a handlful of areas and that's where I noticed some other debug code trying to identify duplicate service registration (which is what I added above). 
Long story short, I should probably revert/close this PR.
If you have a bug/workitem # for the Newtonsoft.json.dll CG issue, I'll use that for a new PR otherwise I'll create an item for it.